### PR TITLE
refactor|chore(node): new TestNode testing utility and node tests cleanup

### PR DESCRIPTION
**Pull request by Wondertan**
Just took my broom and put on my cleaning hat.

### CODE REVIEW

File: node/node_bridge_test.go [hunk 1]
```diff
@@ -42,100 +22,14 @@ func TestBridgeLifecycle(t *testing.T) {
...
-	ctx, cancel = context.WithCancel(context.Background())
-	t.Cleanup(cancel)
-
-	err = node.Stop(ctx)
-	require.NoError(t, err)
-}
-
-// TestBridge_P2P_Streams tests the ability for Bridge nodes to communicate
-// directly with each other via libp2p streams.
-func TestBridge_P2P_Streams(t *testing.T) {
```
Comment (Wondertan) [inline node/node_bridge_test.go:57]: No need to have this test here. Swamp does a good job of ensuring our nodes can talk to each other.

File: node/node_bridge_test.go [hunk 2]
```diff
@@ -8,30 +8,10 @@ import (
...
-	assert.NotZero(t, node.Type)
-}
-
-func TestBridgeLifecycle(t *testing.T) {
-	cfg := DefaultConfig(Bridge)
-	store := MockStore(t, cfg)
-
-	node, err := New(Bridge, store)
-	require.NoError(t, err)
+func TestNewBridgeAndLifecycle(t *testing.T) {
```
Comment (Wondertan) [inline node/node_bridge_test.go:13]: Not sure why we separated those tests initially as they can be simplified into one.

File: node/node_bridge_test.go [hunk 3]
```diff
@@ -42,100 +22,14 @@ func TestBridgeLifecycle(t *testing.T) {
...
-	_, err = stream.Write([]byte("hello"))
-	require.NoError(t, err)
-
-	// stop the connection
-	require.NoError(t, stream.Close())
-}
-
-func TestBridge_WithRemoteCore(t *testing.T) {
-	// TODO(@Wondertan): Fix core
-	t.Skip("Skip until we fix core")
```
Comment (Wondertan) [inline node/node_bridge_test.go:104]: There were some flakiness or something idk, but here as well Swamp tests this for us

File: node/node_bridge_test.go [hunk 4]
```diff
@@ -42,100 +22,14 @@ func TestBridgeLifecycle(t *testing.T) {
...
 	ctx, cancel = context.WithCancel(context.Background())
 	t.Cleanup(cancel)
 
 	err = node.Stop(ctx)
 	require.NoError(t, err)
-	err = remoteCore.Stop()
-	require.NoError(t, err)
-}
-
-func TestBridge_NotPanicWithNilOpts(t *testing.T) {
```
Comment (Wondertan) [inline node/node_bridge_test.go:134]: This is simply useless.

File: node/node_bridge_test.go [hunk 5]
```diff
@@ -154,16 +48,3 @@ func TestBridge_WithMockedCoreClient(t *testing.T) {
 	err = node.Stop(ctx)
 	require.NoError(t, err)
 }
-
-func TestBridge_WithMutualPeers(t *testing.T) {
```
Comment (Wondertan) [inline node/node_bridge_test.go:158]: There is the same test for a light client. There is zero value in testing options for both node types as the code for them is the same. I would just use light client only for testing options.

**Review by liamsi — APPROVED** [review]
We need to add a CI step to see that we are not decreasing test coverage!

**Review by renaynay — APPROVED** [review]
lgtm, nothing sticks out

**Comment by Bidon15** [comment]
Ok. I will start deploying our test-coverage as this is a low-hanging fruit
Ref: https://github.com/celestiaorg/celestia-node/issues/368

**Pull request merged**